tests/codegen/vec-in-place.rs

```diff
@@ -85,9 +98,18 @@ pub fn vec_iterator_cast_deaggregate_fold(vec: Vec<Baz>) -> Vec<[u64; 4]> {
     // CHECK-NOT: loop
     // CHECK-NOT: call
 
+    // Having a scope call is fine; that's not what this is trying to avoid
+    // CHECK: call void @llvm.experimental.noalias.scope.decl
+
+    // CHECK-NOT: loop
+    // CHECK-NOT: call
+
     // Safety: For the purpose of this test we assume that Bar layout matches [u64; 4].
     // This currently is not guaranteed for repr(Rust) types, but it happens to work here and
     // the UCG may add additional guarantees for homogenous types in the future that would make this
     // correct.
     vec.into_iter().map(|e| unsafe { std::mem::transmute(e) }).collect()
 }
+
+// Keep the `call` above from matching this not-really-a-call
+// CHECK: Function Attr{{.+}}nocallback
```

Comment:
probably better to add a `CHECK: ret` to the function above. I left that out when writing the test since it happened to work at the time.